website/docs/operations.md

```diff
@@ -0,0 +1,164 @@
+any violations it discovers. To limit traffic to the API server and to avoid contention writing audit results
+to constraints, audit should run as a singleton pod.
+
+### Required Behaviors
+
+At a high level, this requires:
+
+* Listing all objects on the cluster to scan them for violations
+* Ingesting constraint templates
+* Creating CRDs for a corresponding constraint template
+* Ingesting constraints
+* Reporting the status of ingested constraints/templates
+* Watching and syncing resources specified by the `Config` resource to support referential constraints
+* Writing audit results back to constraints (subject to a cap on # of results per constraint)
+
+### Permissions Required
+
+* The ability to read all objects in the cluster (this can be scoped down if you are not interested in auditing/syncing all objects)
+* The ability to read all `ConstraintTemplate` objects
+* The ability to create CRDs (unfortunately RBAC doesn't have the syntax to scope this down to just CRDs in the `constraints.gatekeeper.sh` group)
+* The ability to write to all `Constraint` resources (members of the group `constraints.gatekeeper.sh`)
+* The ability to create `ConstraintTemplatePodStatus` objects in Gatekeeper's namespace
+* The ability to create `ConstraintPodStatus` objects in Gatekeeper's namespace
+* The ability to write to the `Config` object in Gatekeeper's namespace
+* If you have events enabled, you will need permissions to create events in Gatekeeper's namespace
+
+## Status
+
+__--operation key:__ `status`
+
+Gatekeeper uses an emergent consensus model, where individual pods do not need to talk with each other
+in order to provide its functionality. This allows for scalability, but means we should not write status
+to resources directly due to the risk of write contention, which could increase network traffic exponentially
+relative to the number of pods. Instead, each pod gets its own, private status resource that it alone writes
+to. The Status operation aggregates these status resources and writes them to the status field of the appropriate
+object for the user to consume. Without this operation, the `status` field of constraints and constraint templates
+would be blank.
+
+In order to do its job (eliminating write contention) effectively, the Status operation should be run as a
+singleton.
+
+### Required Behaviors
+
+At a high level, this requires:
+
+* Reading the Constraint[Template]PodStatus resources
+* Writing aggregated results to the `status` fields of constraints/templates
+
+### Permissions Required
+
+* The ability to write to all `ConstraintTemplate` objects
+* The ability to write to all `Constraint` resources (members of the group `constraints.gatekeeper.sh`)
+* The ability to read `ConstraintTemplatePodStatus` objects in Gatekeeper's namespace
+* The ability to read `ConstraintPodStatus` objects in Gatekeeper's namespace
+
+## Mutation Status
+
+__--operation key:__ `mutation-status`
```

Comment:
just realized this is not in https://github.com/open-policy-agent/gatekeeper/blob/master/deploy/gatekeeper.yaml#L2418 I can open a PR for this later.

Reply:
Oh wow, that's a good catch.